contracts/contracts/rem.system/src/delegate_bandwidth.cpp

```diff
@@ -297,9 +296,13 @@ namespace eosiosystem {
 
       if(is_block_producer(from))
       {
-          del_bandwidth_table     del_tbl( _self, from.value );
-          const auto& del = del_tbl.get( from.value, "user has no resources" );
-          check( current_time_point() > del.staked_time + stake_lock_period, "producer cannot undelegate bandwidth during 180 days");
+          const auto &vot = _voters.get(from.value, "user has no resources");
+          _voters.modify(vot, from, [&](auto &v) {
+              check(v.locked_stake >= unstake_quantity.amount, "cannot undelegate more than was staked");
```

Comment:
@AndriiAndrusyshyn please move this check out of modification lambda and change error message in the second one to "cannot undelegate during stake lock period"